src/test/PoolModifyPositionTest.sol

```diff
@@ -45,32 +41,19 @@ contract PoolModifyPositionTest is ILockCallback {
 
         BalanceDelta delta = manager.modifyPosition(data.key, data.params, data.hookData);
 
-        if (delta.amount0() > 0) {
-            if (data.key.currency0.isNative()) {
-                manager.settle{value: uint128(delta.amount0())}(data.key.currency0);
-            } else {
-                IERC20Minimal(Currency.unwrap(data.key.currency0)).transferFrom(
-                    data.sender, address(manager), uint128(delta.amount0())
-                );
-                manager.settle(data.key.currency0);
-            }
-        }
-        if (delta.amount1() > 0) {
-            if (data.key.currency1.isNative()) {
-                manager.settle{value: uint128(delta.amount1())}(data.key.currency1);
-            } else {
-                IERC20Minimal(Currency.unwrap(data.key.currency1)).transferFrom(
-                    data.sender, address(manager), uint128(delta.amount1())
-                );
-                manager.settle(data.key.currency1);
-            }
-        }
-
-        if (delta.amount0() < 0) {
-            manager.take(data.key.currency0, data.sender, uint128(-delta.amount0()));
-        }
-        if (delta.amount1() < 0) {
-            manager.take(data.key.currency1, data.sender, uint128(-delta.amount1()));
+        (,,, int256 delta0) = _fetchBalances(data.key.currency0, data.sender);
+        (,,, int256 delta1) = _fetchBalances(data.key.currency1, data.sender);
+
+        if (data.params.liquidityDelta > 0) {
+            assert(delta0 > 0 || delta1 > 0);
+            assert(!(delta0 < 0 || delta1 < 0));
```

Comment:
this second assert seems unneeded?

Reply:
The second assert is to check that modify position never returns a negative delta (which would be correct for that function. Where the first is to check that at least one of the deltas is non-0

Reply:
Ah yes I see